homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -98,15 +95,15 @@ def lookupstatusfromcode(self, statuscode):
         if self._rs_codes is not None:
             statuscode = str(statuscode).strip()
             for i in self._rs_codes:
-                val = str(self._rs_codes[i]).strip()
+                val = str(self._rs_codes[i][0]).strip()
                 if ',' in val:
```

Comment:
Do we expect a comma in the codes? According to the config schema we should get a list of codes, ie not comma separated.

Reply:
When I tested I did get a list with 1 item, and in that item were the codes separated by commas. That is why the code is at it is. I do not want to change the config the users have to pass in too much.

Reply:
Ok, change config schema to use `cv.ensure_list_csv` instead of `cv.ensure_list`. That will turn the comma separated string into a list for you. Then we don't need to make a breaking change but you avoid having to do all this parsing here.

Reply:
Ok, I made the change and changed the parsing.